src/main/java/gyro/google/compute/HealthCheckResource.java

```diff
@@ -0,0 +1,310 @@
+package gyro.google.compute;
+
+import java.io.IOException;
+import java.util.Set;
+
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.util.Data;
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.model.HTTP2HealthCheck;
+import com.google.api.services.compute.model.HTTPHealthCheck;
+import com.google.api.services.compute.model.HTTPSHealthCheck;
+import com.google.api.services.compute.model.HealthCheck;
+import com.google.api.services.compute.model.Operation;
+import com.google.api.services.compute.model.SSLHealthCheck;
+import com.google.api.services.compute.model.TCPHealthCheck;
+import gyro.core.GyroException;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Resource;
+import gyro.core.scope.State;
+
+/**
+ * Creates a global health check resource.
+ *
+ * Examples
+ * --------
+ *
+ * .. code-block:: gyro
+ *
+ *      google::compute-health-check health-check-example-basic-http
+ *          name: "http-basic"
+ *
+ *          http-health-check
+ *              request-path: "/myapp"
+ *          end
+ *      end
+ *
+ * .. code-block:: gyro
+ *
+ *      google::compute-health-check health-check-example-advanced-https
+ *          check-interval-sec: 30
+ *          description: "The description goes here."
+ *          healthy-threshold: 8
+ *          name: "https-advanced"
+ *          timeout-sec: 29
+ *          unhealthy-threshold: 6
+ *
+ *          https-health-check
+ *              request-path: "/myapp"
+ *              response: "okay"
+ *          end
+ *      end
+ *
+ * .. code-block:: gyro
+ *
+ *      google::compute-health-check health-check-example-advanced-http2
+ *          check-interval-sec: 30
+ *          description: "The description goes here."
+ *          healthy-threshold: 8
+ *          name: "http2-advanced"
+ *          timeout-sec: 29
+ *          unhealthy-threshold: 6
+ *
+ *          http2-health-check
+ *              proxy-header: "PROXY_V1"
+ *              request-path: "/myapp"
+ *              response: "okay"
+ *          end
+ *      end
+ *
+ * .. code-block:: gyro
+ *
+ *      google::compute-health-check health-check-example-advanced-tcp
+ *          check-interval-sec: 30
+ *          description: "The description goes here."
+ *          healthy-threshold: 8
+ *          name: "tcp-advanced"
+ *          timeout-sec: 29
+ *          unhealthy-threshold: 6
+ *
+ *          tcp-health-check
+ *              proxy-header: "PROXY_V1"
+ *              request-path: "/myapp"
+ *              response: "okay"
+ *          end
+ *      end
+ *
+ * .. code-block:: gyro
+ *
+ *      google::compute-health-check health-check-example-advanced-ssh
+ *          check-interval-sec: 30
+ *          description: "The description goes here."
+ *          healthy-threshold: 8
+ *          name: "ssh-advanced"
```

Comment:
I guess now that you are providing examples of all the type, mentioning `advanced` is not needed.
```suggestion
 *          name: "example-ssh"
```

Update this for all the cases and examples